app/controllers/candidate_interface/candidate_interface_controller.rb

```diff
@@ -9,8 +9,12 @@ class CandidateInterfaceController < ActionController::Base
 
   private
 
-    def redirect_to_dashboard_if_submitted
-      redirect_to candidate_interface_application_complete_path if current_application.submitted?
+    def redirect_to_dashboard_if_not_amendable
+      if FeatureFlag.active?('edit_application')
+        redirect_to candidate_interface_application_complete_path if current_application.submitted? && current_application.within_amend_period? == false
+      elsif current_application.submitted?
+        redirect_to candidate_interface_application_complete_path
+      end
```

Comment:
We're unable to refactor this to use a guard clause because the behaviour is different when the feature flag is **_on_** so:

- applications that are submitted and within the amend period can be edited (no redirect)

when the feature flag is **_off_**:

- all applications that are submitted cannot be edited (redirect)